src/frontend/src/components/util/apiCalls.js

```diff
@@ -326,10 +330,22 @@ export const getExternalUserGamesApi = async (gameId) => {
   //   ['G', 'G', 'Y', 'Y', 'X']
   // ])
 
-
   return {
     externalUserIds,
     externalUserGamesMap,
     state,
+  };
+};
+
+export const getStatsApi = async (userId) => {
+  try {
+    const res = await axios.get("http://localhost:5002/getStats/" + userId);
+    if (res.status !== 200) {
+      throw new Error(`HTTP ERROR! Status: ${res.status}`);
+    }
+    return res.data;
+  } catch (error) {
+    console.error("Fetch failed", error);
+    throw new Error("Fetch failed", error);
```

Comment:
Can we have a more detailed error message here citing which API the error came from?